File: leetcode/algorithms/1-two-sums.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> myHashMap;
        for (int i = 0; i < nums.size(); ++i) {
            int valMissing = target - nums[i];
            if (myHashMap.find(valMissing) != myHashMap.end()) {
                vector<int> myResult{ myHashMap[valMissing], i };
                return myResult;
            }
            myHashMap.insert({nums[i], i});
        }
        vector<int> myResult{ 0, 0 };
        return myResult;
    }
};

File: leetcode/algorithms/1-two-sums.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        myHash = {}
        i = 0
        for num in nums:
            valMissing = target - num
            if valMissing in myHash:
                return [myHash[valMissing], i]
            myHash[num] = i
            i += 1
        